File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mp = new HashMap<>();

        for(int i = 0; i < nums.length; i++){
            int toFind = target - nums[i];
            if(mp.containsKey(toFind)){
                return new int[]{mp.get(toFind), i};
            }
            mp.put(nums[i], i);
        }

        return new int[] {-1,-1};
    }
}

File: Data Structures & Algorithms/single-number/submission-0.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        mp = {}

        for num in nums:
            mp[num] = 1 + mp.get(num, 0)

        for num in mp:
            if mp.get(num, 0) == 1:
                return num

        return -1